### ISSUE TITLE
Proposing a prototype branch behaves incorrectly

### ISSUE THREAD

**Issue by sergej-koscejev** [git-town/git-town#4109]
### Steps to reproduce

```
$ gt prototype
$ gt propose
```

### Command output

```shell
$ gt propose
[my-feature-branch] git fetch --prune --no-tags
...

Looking for proposal online ... none

[my-feature-branch] git checkout main
Switched to branch 'main'
Your branch is up to date with 'origin/main'.

[main] git rebase origin/main --no-update-refs
Current branch main is up to date.

[main] git checkout my-feature-branch
Switched to branch 'my-feature-branch'

[my-feature-branch] git rebase main --no-update-refs
Current branch my-feature-branch is up to date.

open https://github.com/myrepo/myrepo/...?expand=1


branch "my-feature-branch" is no longer a prototype branch
```

### Additional information

It seems that the proposal commands run first but the branch is not pushed because it's a prototype branch. Then the branch is made non-prototype at the end but that's too late. But a second `gt propose` works afterwards.

**Comment by kevgo** [comment]
I can confirm this bug. We have end-to-end tests for this ([1](https://github.com/git-town/git-town/blob/main/features/propose/features/branch_types/prototype.feature), [2](https://github.com/git-town/git-town/blob/main/features/propose/features/branch_types/prototype_child_branch.feature)) but they miss that the ex-prototype branch doesn't get pushed. 😆 

The root cause is that the branch has prototype status while Git Town assembles the program for proposing the branch. So that's going to be a program for syncing the branch as a prototype branch. The prototype status then gets removed when this program runs.

A possible solution is to add an opcode to the propose program that pushes the ex-prototype branch to origin in case it is still a local branch.

**Comment by kevgo** [comment]
This is now shipped in [Git Town 16.5](https://github.com/git-town/git-town/releases/tag/v16.5.0).
